cbits/itoa.c

```diff
@@ -213,3 +215,79 @@ char* _hs_bytestring_long_long_uint_hex (long long unsigned int x, char* buf) {
     }
     return next_free;
 };
+
+// unsigned ints (32 bit words)
+void _hs_bytestring_builder_uint_fixed_width_hex (int width,
+                                                  unsigned int x,
+                                                  char* buf) {
+    while (--width >= 0) {
+      buf[width] = digits[x & 0xf];
+      x >>= 4;
+    }
+};
+
+// unsigned ints (64 bit words)
+void _hs_bytestring_builder_long_long_uint_fixed_width_hex (int width,
+                                                            long long unsigned int x,
+                                                            char* buf) {
+    while (--width >= 0) {
+      buf[width] = digits[x & 0xf];
+      x >>= 4;
+    }
+};
+
+// unsigned ints (32 bit words)
+char* _hs_bytestring_uint_hex_upper (unsigned int x, char* buf) {
+    // write hex representation in reverse order
+    char c, *ptr = buf, *next_free;
+    do {
+        *ptr++ = digits_upper[x & 0xf];
+        x >>= 4;
+    } while ( x );
+    // invert written digits
+    next_free = ptr--;
+    while(buf < ptr) {
+        c      = *ptr;
+        *ptr-- = *buf;
+        *buf++ = c;
+    }
+    return next_free;
+};
+
+// unsigned long ints (64 bit words)
+char* _hs_bytestring_long_long_uint_hex_upper (long long unsigned int x, char* buf) {
```

Comment:
```uint64_t``` means that it requires C99 (or later) and a byte oriented target that has a type that is exactly 64 bits, otherwise it may not exist.

Then there is the issue that it should match the Haskell side, and while there is a  ``CLLong`` type (which represents a C ``long long`` type), there seems to be nothing in ``Foreign.C.Types`` that corresponds to a ``uint64_t``.